plugins/module_utils/foreman_helper.py

```diff
@@ -211,6 +211,35 @@ def find_resources_by_name(self, resource, names, **kwargs):
     def find_resources_by_title(self, resource, titles, **kwargs):
         return [self.find_resource_by_title(resource, title, **kwargs) for title in titles]
 
+    def find_operatingsystem(self, name, params=None, failsafe=False, thin=None):
+        search = 'title~"{}"'.format(name)
+        list_params = {}
+        if params is not None:
+            list_params.update(params)
+        if thin is None:
+            thin = self._thin_default
+        list_params['thin'] = thin
+        results = self.list_resource('operatingsystems', search, list_params)
+        result = None
+        unique_str = ''
+        if len(results) == 1:
+            result = results[0]
+        elif len(results) >= 1:
+            # return the exact match (if available)
+            result = next((result for result in results if result['title'] == name), None)
```

Comment:
funfact, I think this would not work if `openratingsystems` would support the `thin` parameter (they do not) as `title` is not returned for thin (for hosts: https://github.com/theforeman/foreman/blob/f3e9f1f0308fb883637b0a57cbe014ef9f287ec3/app/controllers/api/v2/hosts_controller.rb#L47-L52)

Reply:
I guess, that means the other solution is future-thin-proof?
I'll change it due to majority request.

Reply:
It's more future proof by the definition in https://twitter.com/lunasorcery/status/1156650841216049152, yes ;-)